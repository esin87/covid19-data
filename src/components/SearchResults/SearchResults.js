import React from 'react';
import LastUpdated from '../LastUpdated/LastUpdated';
import Table from 'react-bootstrap/Table';
import Alert from 'react-bootstrap/Alert';

const SearchResults = ({ data, lastSearch, formatNumber }) => {
	if (!data.length) {
		return (
			<Alert variant='warning' style={{ marginTop: '2.5rem' }}>
				Sorry, no data found for "{lastSearch}". Please try another query.
			</Alert>
		);
	}

	const latestData = data[data.length - 1];

	return (
		<div className='data-container' style={{ marginTop: '2.5rem' }}>
			<h3>{latestData.Country}</h3>
			<Table>
				<thead>
					<tr>
						<th></th>
						<th>Confirmed cases</th>
						<th>Deaths</th>
						<th>Recovered</th>
					</tr>
				</thead>
				<tbody>
					<tr>
						<td>Total</td>
						<td>{formatNumber(latestData.Confirmed)}</td>
						<td>{formatNumber(latestData.Deaths)}</td>
						<td>{formatNumber(latestData.Recovered)}</td>
					</tr>
				</tbody>
			</Table>
			<LastUpdated date={latestData.Date} />
		</div>
	);
};

export default SearchResults;
