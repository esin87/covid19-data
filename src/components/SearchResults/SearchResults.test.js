import React from 'react';
import SearchResults from './SearchResults';
import { mount, render } from 'enzyme';

describe('SearchResults component', () => {
	let component;
	const lastSearch = 'United States';
	function formatNumber(num) {
		return num.toString().replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,');
	}
	const data = [
		{
			Country: 'Switzerland',
			CountryCode: 'CH',
			Lat: '46.82',
			Lon: '8.23',
			Confirmed: 20505,
			Deaths: 666,
			Recovered: 6415,
			Active: 13424,
			Date: '2020-04-05T00:00:00Z',
			LocationID: '628d4f12-6ebe-4fa9-b046-77ff0b894a4e',
		},
	];

	beforeEach(() => {
		component = mount(
			<SearchResults
				data={data}
				lastSearch={lastSearch}
				formatNumber={formatNumber}
			/>
		);
	});

	it('renders', () => {
		render(component);
	});

	it('should match the snapshot', () => {
		expect(component.html()).toMatchSnapshot();
	});

	it('should have an h3 with appropriate text', () => {
		expect(component.find('h3').text()).toBe('Switzerland');
	});

	it('should have a muted text class with appropriate text', () => {
		expect(component.find('.text-muted').at(0).text()).toBe(
			'Last updated: a month ago'
		);
	});

	it('receives the appropriate props', () => {
		expect(component.props()).toEqual({
			lastSearch: 'United States',
			formatNumber: expect.any(Function),
			data: [
				{
					Country: 'Switzerland',
					CountryCode: 'CH',
					Lat: '46.82',
					Lon: '8.23',
					Confirmed: 20505,
					Deaths: 666,
					Recovered: 6415,
					Active: 13424,
					Date: '2020-04-05T00:00:00Z',
					LocationID: '628d4f12-6ebe-4fa9-b046-77ff0b894a4e',
				},
			],
		});
	});
});
