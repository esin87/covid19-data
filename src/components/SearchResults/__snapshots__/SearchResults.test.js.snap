// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component should match the snapshot 1`] = `"<div class=\\"data-container\\" style=\\"margin-top: 2.5rem;\\"><h3>Switzerland</h3><table class=\\"table\\"><thead><tr><th></th><th>Confirmed cases</th><th>Deaths</th><th>Recovered</th></tr></thead><tbody><tr><td>Total</td><td>20,505</td><td>666</td><td>6,415</td></tr></tbody></table><p class=\\"text-muted\\" style=\\"margin-top: 0.5rem;\\">Last updated: <time datetime=\\"1586044800000\\">a month</time> ago</p></div>"`;
